agasc/agasc.py

```diff
@@ -1,42 +1,178 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 import os
 from pathlib import Path
+import warnings
+from contextlib import ContextDecorator
 
 import numpy as np
 import numexpr
 import tables
-import warnings
 
 from Chandra.Time import DateTime
 from astropy.table import Table, Column
-from ska_helpers.utils import LazyDict
+from ska_helpers.utils import lru_cache_timed
+
+__all__ = ['sphere_dist', 'get_agasc_cone', 'get_star', 'get_stars',
+           'MAG_CATID_SUPPLEMENT', 'BAD_CLASS_SUPPLEMENT', 'get_supplement_table',
+           'disable_supplement']
+
+DISABLE_SUPPLEMENT_ENV = 'AGASC_DISABLE_SUPPLEMENT'
+MAG_CATID_SUPPLEMENT = 100
+BAD_CLASS_SUPPLEMENT = 100
+
+RA_DECS_CACHE = {}
+COMMON_DOC = """\
+If ``use_supplement`` is ``True`` and there is no ``AGASC_DISABLE_SUPPLEMENT``
+    environment variable set, then stars with available mag estimates or bad
+    star entries in the AGASC supplement are updated in-place in the output
+    ``stars`` Table:
+
+    - ``MAG_ACA`` and ``MAG_ACA_ERR`` are set according to the supplement.
+    - ``MAG_CATID`` (mag catalog ID) is set to ``MAG_CATID_SUPPLEMENT`` (100).
+    - If ``COLOR1`` is 0.7 or 1.5 then it is changed to 0.69 or 1.49 respectively.
+    - If ``AGASC_ID`` is in the supplement bad stars table then CLASS is set to
+      ``100 + <source>``, where ``<source>`` is the bad star source identifier.
+
+    The default ``agasc_file`` is ``<AGASC_DIR>/miniagasc.h5``, where
+    ``<AGASC_DIR>`` is either the ``AGASC_DIR`` environment variable if defined
+    or ``$SKA/data/agasc``."""
+
+
+def default_agasc_dir():
+    """Path to the AGASC directory.
+
+    This returns the ``AGASC_DIR`` environment variable if defined, otherwise
+    ``$SKA/data/agasc``.
+
+    :returns: Path
+    """
+    if 'AGASC_DIR' in os.environ:
+        out = Path(os.environ['AGASC_DIR'])
+    else:
+        out = Path(os.environ['SKA'], 'data', 'agasc')
+    return out
+
+
+def default_agasc_file():
+    """Default main AGASC file ``agasc_dir() / miniagasc.h5``.
+
+    :returns: str
+    """
+    return str(default_agasc_dir() / 'miniagasc.h5')
 
 
-__all__ = ['sphere_dist', 'get_agasc_cone', 'get_star', 'get_stars', 'MAG_CATID_SUPPLEMENT']
+@lru_cache_timed(timeout=3600)
+def get_supplement_table(name, agasc_dir=None, as_dict=False):
```

Comment:
Originally, I thought this function would be in the supplement namespace. I.e. `agasc.supplement.get_table` instead of `agasc.get_supplement_table`

Reply:
I can do that, maybe like this:
- Move `get_supplement_table` into its own module in `agasc/supplement/utils.py` (better name?). There is no need to load all the other modules that all relate to maintaining the supplement rather than user access.
- Factor out the path definitions so that `default_agasc_dir` is available to `agasc/supplement/utils.py` without a circular import. Putting paths into a separate `paths.py` module could be a good thing anyway if it gets used elsewhere in the package.
- Update the package `__init__.py` to import the function from `agasc.supplement.utils`.

Reply:
sounds good to me.

Reply:
Done.